### DISCUSSION TITLE
Looking for SASL Users

### DISCUSSION THREAD

**Discussion by petergoldstein** [petergoldstein/dalli#842] [Announcements]
Do you use Dalli with SASL?  If so, I'd like to hear from you.

The SASL tests have been disabled in Dalli for a very long time.  This effectively makes it impossible confirm that Dalli is working with a SASL configured memcached instance.  And having untested features whose operation is not verified by CI makes the system more difficult to support.

This problem is made more difficult by a few things:

- SASL support is no longer standard on a Homebrew installed memcached
- SASL is only supported for binary protocol operation, and the binary protocol is now "deprecated".  This effectively means that SASL support is no longer a focus of the memcached team.
- There are comments in the code indicating that it may never have worked with local instances - but only with Couchbase instances accessed from Heroku.  This adds uncertainty to the state of the code.
- I've had some difficulty getting SASL support working on an EC2 box.  There's more to try here, but my hope that it would just "work" hasn't paid off.

My ideal scenario would be to get SASL tests working in CI for the 3.x version, and then have a larger discussion about whether to continue that support going forward to 4.x.

If you have memcached with SASL running in your environment, and especially if you're using Dalli with it, I'd love your help to get the tests running again.  And to understand whether this is an important feature for you going forward.  Please reach out in the comments.

Thanks!

**Comment by apiology** [comment]
My SASL-equipped memcached instance comes from when I spun up a free Heroku instance, which allowed me to add on a small, free "Memcached Cloud" server (actually hosted via https://app.redislabs.com/).  You can export the creds and connect from anywhere, so you could presumably spin up one and use it for CI purposes.

**Comment by ryanfb** [comment]
This `memcached-sasl` Dockerfile/image seems to work in my local testing: https://github.com/vincentsarago/memcached-sasl / https://cloud.docker.com/repository/docker/remotepixel/memcached-sasl

I've made a demo repo with `docker-compose` here: https://github.com/ryanfb/dalli-memached-sasl-demo

**Comment by petergoldstein** [comment]
@ryanfb and @apiology thanks so much.  The Dockerfile was especially helpful in figuring out how to get this running in our current CI.

I may wind up switching to a Docker based memcached installation in the future, but for now I'm going to stick with the install script.  It works, even if it's not quite as pretty.

**Reply by jamiemccarthy** [reply]
@petergoldstein Are you still interested in moving to docker-based CI?

**Reply by petergoldstein** [reply]
@jamiemccarthy Yes, I'd much rather use Dockerized memcached services in preference to the current install script.  There are some challenges with ports and checking different memcached configurations (see the specs) but I think that's be cleaner if those issues can be addressed
